server/controllers/studentControllers/studentSearchRecommend.js

```diff
@@ -78,17 +82,17 @@ const searchProjects = async (req, res) => {
         const decodeAccessToken = JWT.verify(accessToken, process.env.SECRET_ACCESS_TOKEN);
 
         const promises = [
-            User.findOne({ email: decodeAccessToken.email }),
+            retrieveOrCacheUsers(req, decodeAccessToken.email),
             Project.aggregate(pipeline)
         ]
-        
+
         await Promise.all(promises).then(results => {
             student = results[0];
             searchResults = results[1]
         });
 
-        //check if user type is a student
-        if (student.userType.Type != process.env.STUDENT) { return res.status(401).json(generateRes(false, 401, "ACCESS_DENIED", { details: "Invalid account type for this operation." })); }
+        //check if user type is a student or faculty, will probably remove
+        // if (student.userType.Type != process.env.STUDENT || student.userType.Type != process.env.FACULTY) { return res.status(401).json(generateRes(false, 401, "ACCESS_DENIED", { details: "Invalid account type for this operation." })); }
```

Comment:
delete code comment?

Reply:
No, its there to allow all users to access the search route. We haven't had a discussion on who should have access to that route so Ieft it commented until it gets sorted then it will be removed or kept.